lib/modules/manager/devcontainer/extract.ts

```diff
@@ -0,0 +1,95 @@
+import { logger } from '../../../logger';
+import { isValidDependency } from '../custom/regex/utils';
+import { getDep as getDockerDep } from '../dockerfile/extract';
+import type {
+  ExtractConfig,
+  PackageDependency,
+  PackageFileContent,
+} from '../types';
+import { DevContainerFile } from './schema';
+
+export function extractPackageFile(
+  content: string,
+  packageFile: string,
+  extractConfig: ExtractConfig,
+): PackageFileContent | null {
+  try {
+    const file = DevContainerFile.parse(content);
+    const deps: PackageDependency[] = [];
+
+    const image = file?.image ?? null;
+    const imageDep = getDep(image, packageFile, extractConfig.registryAliases);
+
+    if (imageDep) {
+      deps.push(imageDep);
+    } else {
+      logger.debug(
+        { packageFile },
+        'No image defined in dev container JSON file.',
+      );
+    }
+
+    const features = file.features ?? {};
+
+    if (features) {
+      for (const feature of Object.keys(features)) {
+        const featureDep = getDep(
+          feature,
+          packageFile,
+          extractConfig.registryAliases,
+        );
+        if (!featureDep) {
+          logger.debug(
+            { feature, packageFile },
+            'Skipping invalid dependency in dev container JSON file.',
+          );
+          continue;
+        }
+        deps.push(featureDep);
```

Comment:
invert for simplicity
```suggestion
        if (featureDep) {
          deps.push(featureDep);
        }
        logger.debug(
          { feature, packageFile },
          'Skipping invalid dependency in dev container JSON file.',
        );
```